surch/organization.py

```diff
@@ -19,199 +19,200 @@
+        response = requests.get(constants.GITHUB_API_URL.format(
+            self.item_type, self.organization), auth=self.git_credentials)
+        if response.status_code == requests.codes.NOT_FOUND:
+            self.logger.error(
                 'The organization or user {0} could not be found. '
                 'Please make sure you use the correct type (org/user).'.format(
                     self.organization))
             sys.exit(1)
-        repository_number = \
-            repository_data.json()['{0}_repos'.format(repository_type)]
-        last_page_number = repository_number / repository_per_page
-        if (repository_number % repository_per_page) > 0:
+        return response.json()
+
+    def _get_repos_list(self, repos_per_page, page_num):
+        response = requests.get(constants.REPO_DETAILS_API_URL.format(
+            self.item_type,
+            self.organization,
+            'public',
+            repos_per_page,
+            page_num), auth=self.git_credentials)
+        return response.json()
+
+    def _parse_repo_data(self, repo_data):
+        return [dict((key, data[key]) for key in ['name', 'clone_url'])
+                for data in repo_data]
+
+    def _get_repos_data(self, repos_per_page=100):
+        self.logger.info(
+            'Retrieving repository information for the {0}...'.format(
+                'organization' if self.is_organization else 'user'))
+        org_data = self._get_org_data()
+        repo_count = org_data['public_repos']
+        last_page_number = repo_count / repos_per_page
+        if (repo_count % repos_per_page) > 0:
             # Adding 2 because 1 for the extra repos that mean more page,
             #  and 1 for the next for loop.
             last_page_number += 2
-
+            repos_data = []
             for page_num in range(1, last_page_number):
-                repository_data = requests.get(
-                    REPO_DETAILS_API_URL.format(self.item_type,
-                                                self.organization,
-                                                repository_type,
-                                                repository_per_page,
-                                                page_num), auth=auth)
-
-                for repository in repository_data.json():
-                    self.repository_data.append(repository)
-                self.repository_specific_data = \
-                    self._parse_json_list_of_dict(['name', url_type])
-
-    def _parse_json_list_of_dict(self, list_of_arguments):
-        return [dict((key, data[key]) for key in list_of_arguments)
-                for data in self.repository_data]
-
-    def search(self, search_list, url_type='clone_url'):
+                repo_data = self._get_repos_list(repos_per_page, page_num)
+                repos_data.extend(self._parse_repo_data(repo_data))
+            return repos_data
+
+    def get_include_list(self, repos_data, include=None, exclude=None):
```

Comment:
since include and exclude cannot work together I suggest to make sure that this function will test for this condition. otherwise there's no problem to run it with both include and exclude.